src/main/java/de/tum/in/www1/artemis/domain/Submission.java

```diff
@@ -110,12 +112,48 @@ public Result getLatestResult() {
         return null;
     }
 
+    /**
+     * Used to get result by correction round
+     * @param correctionRound to get result by
+     * @return the result based on the given correction-round
+     */
+    @Nullable
+    @JsonIgnore
+    public Result getResultByCorrectionRound(Long correctionRound) {
+        if (results != null && results.size() > correctionRound) {
+            return results.get(correctionRound.intValue());
+        }
+        return null;
+    }
+
+    /**
+     * Used to get result by correction round ignoring all automatic results
+     * @param correctionRound to get result by
+     * @return the result based on the given correction-round
+     */
+    @Nullable
+    @JsonIgnore
+    public Result getResultByCorrectionRoundIgnoreAutomatic(Long correctionRound) {
+        List<Result> withoutAutomaticResults = results.stream().filter(result -> result != null && !result.getAssessmentType().equals(AssessmentType.AUTOMATIC))
```

Comment:
Not sure but I think you can omit `result != null` because you can only apply the filter condition is only applied on actual elements

Reply:
yep you are right :) here we actually do not need it, it was removed.